client/state/ui/billing-transactions/reducer.js

```diff
@@ -0,0 +1,70 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import {
+	BILLING_TRANSACTIONS_FILTER_SET_APP,
+	BILLING_TRANSACTIONS_FILTER_SET_MONTH,
+	BILLING_TRANSACTIONS_FILTER_SET_NEWEST,
+	BILLING_TRANSACTIONS_FILTER_SET_PAGE,
+	BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+} from 'state/action-types';
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+
+/**
+ * Returns the updated app filter state after an action has been dispatched
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {String}        Updated state
+ */
+export const app = createReducer( null, {
```

Comment:
All the reducers here are pretty simple; could probably benefit if we avoid adding unnecessary overhead by using `createReducer` and converting them to simple functions with `switch` calls instead.